engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -614,7 +615,16 @@ protected void serviceStart() throws Exception
       }
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      this.appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      String host = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+
+      // For backward compatibility, not adding scheme in TrackingURL for non-HTTPS
+      // TODO: Remove the check in next major release and add scheme always
+      if (ConfigUtils.isSSLEnabled(config)) {
+        String scheme = ConfigUtils.getSchemePrefix(config);
```

Comment:
@gauravgopi123 @vrozov I had same thing in mind, when I introduced isSSLEnabled function. And it was basically introduced to add scheme conditionally and to avoid compatibility issues before major release.
As you already got an idea, these changes were made while keeping future changes in mind. Ultimately we would like to see something like:

`      this.appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
      this.appMasterTrackingUrl = ConfigUtils.getSchemePrefix(config) + this.appMasterTrackingUrl;`

So IMO, we can make changes as below. And during next major release simply remove if condition.

`      this.appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();

      // For backward compatibility, not adding scheme in TrackingURL for non-HTTPS
      // TODO: Remove the check in next major release and add scheme always
      if (ConfigUtils.isSSLEnabled(config)) {
        this.appMasterTrackingUrl = ConfigUtils.getSchemePrefix(config) + this.appMasterTrackingUrl;
      }`

Please let me know your thoughts on this.